rig-lancedb/src/utils/mod.rs

```diff
@@ -0,0 +1,28 @@
+pub mod deserializer;
+
+use deserializer::RecordBatchDeserializer;
+use futures::TryStreamExt;
+use lancedb::query::ExecutableQuery;
+use rig::vector_store::VectorStoreError;
+
+use crate::lancedb_to_rig_error;
+
+/// Trait that facilitates the conversion of columnar data returned by a lanceDb query to serde_json::Value.
+/// Used whenever a lanceDb table is queried.
+pub trait Query {
```

Comment:
It's only public in the crate (in `lib.rs`, I define `mod utils` instead of `pub mod utils`). But yeah, it needs to be public in the crate bc it is accessed from outside that file